### ISSUE TITLE
Failing with SIGSYS on OpenBSD 7.3

### ISSUE THREAD

**Issue by Liorst4** [jart/cosmopolitan#826]
Tried `redbean.com` and some of the bootstrap files in the `build` directory, got an error about a bad syscall.
using `ktrace` and `kdump` reveled that the syscall in question is number 197 (0xc5): `pad_mmap`

Looks like a few months ago it was made obsolete.
https://github.com/openbsd/src/commit/8c7f5cc47d34f1bd83a08278dad544a34184dabf

The current number for `mmap` is 49 (0x31).
https://github.com/openbsd/src/blob/92190ee3632a7848abd5d7783e1dc3c5cef744ee/sys/kern/syscalls.master#L129

Looks like golang is facing the same issue https://github.com/golang/go/issues/59661

Where the mmap syscall number is set in the code base:
https://github.com/jart/cosmopolitan/blob/72f8bd10b7391f0dfa8cf454eabee19dce4b7fc7/libc/sysv/consts.sh#LL1797C7-L1797C7

**Comment by neezer** [comment]
I'm getting this too, I think.

```
# downloaded https://redbean.dev/redbean-2.2.com
# chmod +x redbean-2.2.com

$ ktrace -tc ./redbean-2.2.com
Bad system call (core dumped)

$ kdump -f ktrace.out
# ...
 33721 .ape     RET   execve JUSTRETURN
 33721 .ape     CALL  access(0x203020,0x1<X_OK>)
 33721 .ape     RET   access 0
 33721 .ape     CALL  open(0x203020,0<O_RDONLY>)
 33721 .ape     RET   open 3
 33721 .ape     CALL  read(3,0x202000,0x1000)
 33721 .ape     RET   read 4096/0x1000
 33721 .ape     CALL  #197 (obsolete pad_mmap)()

$ uname -a
OpenBSD atlas.home 7.3 GENERIC.MP#1125 amd64
```

**Comment by jart** [comment]
I can fix the `off_t` system calls. However we're going to need to bump our OpenBSD minimum supported version to 7.3. I spent a few hours looking into how we could maintain backwards compatibility with OpenBSD 6.4. Sadly it was just too ugly to be worth doing. OpenBSD shouldn't have changed the system call ordinals when making this change, since it's such a clean and easily supportable ABI change (with the exception of lseek, which requires argument reordering).

OpenBSD 7.3 also seems designed to prohibit mixing `.text` and `.rodata`. If I do something like `write(2, "hello", 5)` and the linker placed that string literal in an ELF program header with the `PF_X` bit set, then the kernel will return `EFAULT`. Supporting that will likely require adding ~4kb of footprint to each binary, similar to the ~4kb we're already paying for the `.privileged` segment which makes code morphing possible without violating OpenBSD's `W^X` invariant.

I'm hoping to have this ready in the next few hours. **You're going to need to update your APE loader to version 1.1.** I'll provide instructions for doing that once I'm finished releasing it. I'll also cut a new release of Redbean soon.

**Comment by jart** [comment]
This issue should now be resolved by 40eb3b9d5d7df65181c0b229282120443cdbec64. This change not only fixes OpenBSD but offers improvements for other platforms too. Please read the commit for further details.

A new APE Loader v1.1 has been introduced, which should be both forwards and backwards compatible. If you encounter an error like this:

```
ape error: ./a64l_test.com: APE phdrs must be 4096-aligned and 4096-padded
```

Then it probably means you need to upgrade your systemwide install, since `/usr/bin/ape` is unversioned:

```
ape/apeuninstall.sh
ape/apeinstall.sh
```

We didn't need to introduce bloat to make this change. I pulled off enough linker tricks that binaries might actually end up being smaller. We also didn't need to sacrifice backwards compatibility with recent older versions of OpenBSD, since the new ordinals were probably introduced a few years ago. I've confirmed with my test fleet that all of Cosmopolitan's tests pass on both OpenBSD 7.2 and 7.3. New nightlies have been pushed to the website. I've also tested that the new cosmocc toolchain described by the README is working like a charm. Enjoy!
